### PR TITLE
Added unit tests for some of the functions in d_graphs

### PR DESCRIPTION
**Pull request by ayahusseini**
- to increase coverage, i think we'll have to utilise the streamlit module

### CODE REVIEW

File: dashboard/test_d_graphs.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import re
+import altair as alt
+import pytest
+from unittest.mock import patch
+import pandas as pd
+import datetime
+from d_graphs import pivot_df, get_last_point, generate_html, add_source_columns, create_bubble_chart
+
+
+@pytest.fixture
```
Comment (indiah444) [inline dashboard/test_d_graphs.py:12]: Great use of fixtures :)

File: dashboard/test_d_graphs.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
+# pylint skip-file
+"""Testing the d_graphs file"""
+import re
+import altair as alt
+import pytest
+from unittest.mock import patch
+import pandas as pd
+import datetime
```
Comment (indiah444) [inline dashboard/test_d_graphs.py:8]: Small detail, but you could add a space between python libraries and functions you're importing from other scripts.

File: dashboard/test_d_graphs.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
+# pylint skip-file
+"""Testing the d_graphs file"""
+import re
```
Comment (indiah444) [inline dashboard/test_d_graphs.py:3]: Could also add a space here between native and non-native libraries, but small detail!

Comment (ayahusseini) [reply dashboard/test_d_graphs.py:3]: Good catch - will make the changes

**Review by megansam5 — APPROVED** [review]
Nice use of tests!!

**Pull request merged**